cli/Sources/Noora/Components/SingleChoicePrompt.swift

```diff
@@ -131,6 +141,121 @@ struct SingleChoicePrompt {
         return selectedOption.0
     }
 
+    private func runCancelable<T: Equatable>(options: [(T, String)]) -> T? {
```

Comment:
I'd expect this method to reuse the `run` method (or the other way around). Right now, we're duplicating a lot of code, making this piece harder to maintain.